### PR TITLE
Dashboard filtering merge

### PR DESCRIPTION
**Pull request by isaac-looker**
merging dataset selection and filtering logic

- we'll render the actual Plot component better in future PRs

see example behavior here..

[selector-demo.webm](https://github.com/Spelman-College/spelman-dashboard/assets/51137263/2ce1c7b9-ad21-40d9-a887-6f5b084371e7)

current demo dashboard rendering..

current

### CODE REVIEW

**Review by mattsmithtaiko — COMMENTED** [review]
Looking good overall! I notice the dropdown's values don't change when you flip the toggle. Is that coming in a future PR?

File: frontend/src/router/index.ts [hunk 1]
```diff
@@ -68,30 +67,13 @@ const router = createRouter({
...
-      path: '/data/:view/:dataset',
-      component: DataDashboardDatasetView,
-      meta: { reload: true },
-    },
-    {
-      path: '/data/:view',
-      redirect: '/data-dashboard'
-    },
-    {
-      path: '/data',
```
Comment (cernst122) [inline frontend/src/router/index.ts:80]: Can we either keep the `/data` redirect and point to the new welcome view path, or edit our existing dashboard routing? I believe we have buttons on the header, homepage, and individual book views that break here.

**Review by cernst122 — CHANGES_REQUESTED** [review]
Great work! Is the plan to pull in the filtering variable chips in a followup PR as well?

**Comment by isaac-looker** [comment]
> Looking good overall! I notice the dropdown's values don't change when you flip the toggle. Is that coming in a future PR?

fixed

File: frontend/src/router/index.ts (hunk 1 above)
Comment (isaac-looker) [reply frontend/src/router/index.ts:80]: I'm not seeing any links to /data from any .vue file.

**Comment by isaac-looker** [comment]
> Great work! Is the plan to pull in the filtering variable chips in a followup PR as well?

Yes

File: frontend/src/router/index.ts (hunk 1 above)
Comment (cernst122) [reply frontend/src/router/index.ts:80]: [frontend/src/views/Header.vue](https://github.com/Spelman-College/spelman-dashboard/blob/main/frontend/src/views/Header.vue) line 15 uses `router.push` syntax to link to /data. I believe the homepage and book views use this approach as well.

Comment (isaac-looker) [reply frontend/src/router/index.ts:80]: _fixed_

**Review by cernst122 — APPROVED** [review]

**Pull request merged**